app/assets/javascripts/koi/components/form-managed-asset.js

```diff
@@ -5,80 +5,84 @@
-          if(url) {
-            thumbnail.innerHTML = "<img src='" + getThumbnail(url) + "' />";
-            removeButton.parentNode.style.display = "block";
-          } else {
-            thumbnail.innerHTML = "<div class=\"composable--asset-field--empty-image\"></div>"
-            removeButton.parentNode.style.display = "none";
+          // Update the thumbnail placeholder markup
+          function updateThumbnail(url){
+            if(url) {
+              thumbnail.innerHTML = "<img src='" + getThumbnail(url) + "' />";
+              removeButton.parentNode.style.display = "block";
+            } else {
+              thumbnail.innerHTML = "<div class=\"composable--asset-field--empty-image\"></div>"
+              removeButton.parentNode.style.display = "none";
+            }
           }
-        }
 
-        // Open modal
-        function startBrowsing(e){
-          e.preventDefault();
-          var popupOptions = $.extend({}, Ornament.popupOptions, {
+          // Open modal
+          function startBrowsing(e){
+            e.preventDefault();
+            var popupOptions = $.extend({}, Ornament.popupOptions, {
               type: "iframe",
               items: {
-                  src: "/admin/" + assetType + "s/new?callbackFunction=composableFieldImageCallback" + id
+                src: "/admin/" + assetType + "s/new?callbackFunction=composableFieldImageCallback" + id
               }
-          });
-          $.magnificPopup.open(popupOptions);
-        }
+            });
+            $.magnificPopup.open(popupOptions);
+          }
 
-        // Closing modal when finishing
-        function finishedBrowsing(assetId, url) {
-          $.magnificPopup.close();
-          hiddenField.value = assetId;
-          updateThumbnail(url);
-        }
+          // Closing modal when finishing
+          function finishedBrowsing(assetId, url) {
+            $.magnificPopup.close();
+            hiddenField.value = assetId;
+            updateThumbnail(url);
+          }
 
-        // Remove asset action
-        function clearField(e) {
-          e.preventDefault();
-          hiddenField.value = "";
-          updateThumbnail();
-        }
+          // Remove asset action
+          function clearField(e) {
+            e.preventDefault();
+            hiddenField.value = "";
+            updateThumbnail();
+          }
 
-        // button bindings
-        browseButton.removeEventListener("click", startBrowsing);
-        browseButton.addEventListener("click", startBrowsing);
-        removeButton.removeEventListener("click", clearField);
-        removeButton.addEventListener("click", clearField);
-      }());
+          // button bindings
+          browseButton.removeEventListener("click", startBrowsing);
+          browseButton.addEventListener("click", startBrowsing);
+          removeButton.removeEventListener("click", clearField);
+          removeButton.addEventListener("click", clearField);
```

Comment:
I'm worried that this will not work as intended because the function identity changes every time the method is called. Could we please refactor this together.